lib/ansible/plugins/action/reboot.py

```diff
@@ -120,11 +120,12 @@ def get_shutdown_command_args(self, distribution):
         return args.format(delay_sec=self.pre_reboot_delay, delay_min=delay_min, message=reboot_message)
 
     def get_distribution(self, task_vars):
+        # FIXME: only execute the module if we don't already have the facts we need
```

Comment:
I think I looked into using gathered facts but didn't find a clear way to do that.

Reply:
Several of the other plugins do this (see `package` and `service` for crib notes ;) )